scripts/tools/bootstrap.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+# Installs required dependencies for the application
+# Run from the top level directory (i.e. "thunder/")
+echo "Welcome to Thunder development!"
+echo "Setting up your machine to get ready for development..."
+
+# Install tools on Linux
+if [ "$(uname -s)" = "Linux" ]; then
+  # Set up Node.js 9 repository
+  curl -sL https://deb.nodesource.com/setup_9.x | sudo -E bash -
+
+  echo "Installing Java 8, Maven, and Node using apt-get."
+  apt-get install openjdk-8-jdk maven nodejs -y
+
+  if [[ $? -ne 0 ]]; then
+    echo "[ERROR] There was an error installing packages."
+    echo "[ERROR] Make sure you run the script using 'sudo'"
+
+    exit 1
+  fi
+fi
+
+# Install tools on macOS
+if [ "$(uname -s)" = "Darwin" ]; then
+  # Check for Homebrew
+  which -s brew
+  if [[ $? != 0 ]]; then
+    echo "[ERROR] No Homebrew found in your environment!"
+    echo "[ERROR] Please install Java 8, Maven, Node.js, and NPM separately."
+
+    exit 1
+  else
+    echo "Updating Homebrew..."
+    brew update
+
+    echo "Installing Java 8, Maven, and Node using Homebrew."
+    brew tap caskroom/versions
+
+    brew cask install java8
+    brew install maven
```

Comment:
I've had trouble installing things in the past with home brew. Things like root privileges needed or file permission problems. Can you confirm this will work without user intervention or is that something you'll just have to let happen?

Reply:
It's a good point, it should *just work* but we both know there are always problems that come up. I haven't had a chance to test it on macOS yet since I worked on this from a Linux VM, but I will be sure to test it at some point this week and update the script as necessary. Feel free to test it yourself as well and make changes if something is broken.